test/preprocessor.tests.ts

```diff
@@ -696,4 +696,169 @@ public ActionResult Register()
             ]);
         });
     });
+
+    describe("AppDirectives", () => {
+        it("#:package", async () => {
+            const input = `#:package`;
+            const tokens = await tokenize(input);
+
+            tokens.should.deep.equal([
+                Token.Punctuation.Hash,
+                Token.Punctuation.Colon,
+                Token.Keyword.Preprocessor.Package,
+            ]);
+        });
+
+        it("#:package with name", async () => {
+            const input = `#:package Foo.Goo`;
+            const tokens = await tokenize(input);
+
+            tokens.should.deep.equal([
+                Token.Punctuation.Hash,
+                Token.Punctuation.Colon,
+                Token.Keyword.Preprocessor.Package,
+                Token.Identifier.PreprocessorSymbol("Foo"),
+                Token.Punctuation.Dot,
+                Token.Identifier.PreprocessorSymbol("Goo"),
```

Comment:
Was there a specific reason to make the dots separate tokens? It doesn't look like this is how preprocessor symbols are handled, for example:

<img width="90" height="41" alt="Image" src="https://github.com/user-attachments/assets/856d99f0-76a2-4b1f-81aa-7b70a8195240" />